File: cpp/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for (int i = 0; i < nums.size(); i++){
            vector<int>::iterator it = find(nums.begin(), nums.end(), target-nums[i]);
            if (it != nums.end()){
                cout << distance(nums.begin(), it) << endl;
                vector<int> ret = {i, static_cast<int>(distance(nums.begin(), it))};
                return ret;
            }
        }
    }
};
// @lc code=end

File: python3/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
# Version 1: naive
        # for i in range(len(nums)):
        #     temp = target - nums[i]
        #     if temp in nums:
        #         if nums.index(temp) != i:
        #             return [i, nums.index(temp)]
# Version 2: hashmap (dict in Python)
        hashmap = {}
        for index, num in enumerate(nums):
            hashmap[num] = index
        # res = []
        for i in range(len(nums)):
            j = hashmap.get(target - nums[i])
            if j != None and i != j:
                return [i, j]
            
# @lc code=end

